beacon-chain/blockchain/service.go

```diff
@@ -191,21 +192,39 @@ func (s *Service) StartFromSavedState(saved state.BeaconState) error {
 	s.store = store.New(justified, finalized)
 
 	var store f.ForkChoicer
+	fRoot := bytesutil.ToBytes32(finalized.Root)
 	if features.Get().EnableForkChoiceDoublyLinkedTree {
 		store = doublylinkedtree.New(justified.Epoch, finalized.Epoch)
 	} else {
-		store = protoarray.New(justified.Epoch, finalized.Epoch, bytesutil.ToBytes32(finalized.Root))
+		store = protoarray.New(justified.Epoch, finalized.Epoch, fRoot)
 	}
 	s.cfg.ForkChoiceStore = store
+	fs, err := s.cfg.BeaconDB.StateSummary(s.ctx, fRoot)
+	if err != nil {
+		return errors.Wrap(err, "could not get summary of finalized checkpoint")
+	}
+	if fs == nil {
+		return errInvalidNilSummary
+	}
+
+	if err := store.InsertOptimisticBlock(s.ctx, fs.Slot, fRoot, params.BeaconConfig().ZeroHash, justified.Epoch, finalized.Epoch); err != nil {
```

Comment:
I think that inserting the first block should be part of `doublylinkedtree or protoarray.New` function. What's even good with `New` if it doesn't have the first block anyway?

Reply:
Actually i don't know anymore because `SetOptimisticToValid` is not part of it

Reply:
The finalized block which is the roof of ForkChoice can be optimistic